Source/WebCore/dom/DocumentFullscreen.cpp

```diff
@@ -570,7 +573,7 @@ void DocumentFullscreen::didExitFullscreen(CompletionHandler<void(ExceptionOr<vo
     // Get `fullscreenElement()` before `finishExitFullscreen` clears it.
     RefPtr exitedFullscreenElement = fullscreenElement();
     if (RefPtr frame = document().frame())
-        finishExitFullscreen(frame->mainFrame(), ExitMode::Resize);
+        finishExitFullscreen(Ref { frame->mainFrame() }, ExitMode::Resize);
```

Comment:
should call `protectedMainFrame()`

Reply:
This method doesn't exist on Frame.h . Should I make one? If so, should it return a WeakRef or a Ref?

Reply:
```
Ref<Frame> protectedMainFrame() { return mainFrame(); }
Ref<const Frame> protectedMainFrame() const { return mainFrame(); }
```